File: src/ast/ForwardTemplateReference.java
package ast;

import java.io.StringWriter;

public class ForwardTemplateReference extends BaseNode {
    // TODO: Compute inside the Demangler
    public BaseNode Reference;
    private int _index;

    public ForwardTemplateReference(int index) {
        super(NodeType.ForwardTemplateReference);
        _index = index;
    }

    @Override
    public String GetName() {
        return Reference.GetName();
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        Reference.PrintLeft(writer);
    }

    @Override
    public void PrintRight(StringWriter writer) {
        Reference.PrintRight(writer);
    }

    @Override
    public boolean HasRightPart() {
        return Reference.HasRightPart();
    }
}

File: src/de/fearlesstobi/demangler/ast/ForwardTemplateReference.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class ForwardTemplateReference : BaseNode(NodeType.ForwardTemplateReference) {
    // TODO: Compute inside the de.fearlesstobi.demangler.Demangler
    private val reference: BaseNode? = null

    override fun getName(): String? {
        return reference?.getName()
    }

    public override fun printLeft(writer: StringWriter) {
        reference!!.printLeft(writer)
    }

    public override fun printRight(writer: StringWriter) {
        reference!!.printRight(writer)
    }

    public override fun hasRightPart(): Boolean {
        return reference!!.hasRightPart()
    }
}